Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIWebNavigation.mm

```diff
@@ -231,7 +231,12 @@
         // Only listen for when the main frame loads so we don't call this method more than once.
         @"  if (details.frameId !== 0)",
         @"    return",
-        @"  browser.webNavigation.getFrame({ tabId: details.tabId, frameId: 0 }, function(frame) {",
+        @"  browser.webNavigation.getFrame({ tabId: details.tabId, frameId: 0 }, async function(frame) {",
+
+        // Make sure invalid tab/frame IDs vend an error message - use arbitrary frame and tabIds.
+        @"    await browser.test.assertRejects(browser.webNavigation.getFrame({tabId: (details.tabId + 1), frameId: 0}), /tab not found/i)",
+        @"    await browser.test.assertRejects(browser.webNavigation.getFrame({tabId: details.tabId, frameId: 42}), /frame not found/i)",
```

Comment:
These would make more sense as a separate `TEST(WKWebExtensionAPIWebNavigation, Errors)` test like we do for other APIs. There is nothing here that depends on the event firing, other than the tab ID for the frame version but that is easy to get with `tabs.getCurrent()`.

If you don't want to break into another new test, it at least does not need to be nested in the main `browser.webNavigation.getFrame` callback since `details` is at the top level (just need to make `completedListener` `async`).

Reply:
Yeah - this can definitely be in a different test. I just figured that this was a spot that I already had a tabId, a frame ID, and the verification that the frame had loaded.

Reply:
I'll pull this into a different test.